tests/e2e/scripts/cloud-run/deploy.sh

```diff
@@ -326,7 +326,10 @@ function setup() {
     -i "${APIPROXY_IMAGE}"
 
   # Redeploy ESPv2 to update the service config
-  proxy_args="^++^--tracing_sample_rate=0.05++--tracing_outgoing_context=traceparent++--tracing_incoming_context=traceparent"
+  proxy_args="^++^--tracing_sample_rate=0.05"\
+"++--tracing_outgoing_context=traceparent"\
+"++--tracing_incoming_context=traceparent"\
+"++--envoy_xff_num_trusted_hops=3"
```

Comment:
Why hop=3?

Reply:
Our e2e tests are not testing real client ip restrictions. If we were validating that the e2e client's IP was the one being restricted, then the default hop=0 would work.

Instead, our e2e tests set up fake IPs in the xff header, and they validate against the fake IPs. Here is one example testcase:

https://github.com/GoogleCloudPlatform/esp-v2/blob/585e66acc4d22b49a3116ad44d1457559cfd4f0c/tests/e2e/testdata/bookstore/key_restriction_test.json.template#L3-L12

On GKE, this works fine with the default hop=2. On Serverless, Cloud Run guarantees to only append 1 IP to the end (which doesn't happen on GKE). So  use hop=3.

Reply:
I see.  Could you add some comment for this code for better readability?